library/std/src/io/error.rs

```diff
@@ -156,6 +191,62 @@ pub enum ErrorKind {
     /// [`Ok(0)`]: Ok
     #[stable(feature = "rust1", since = "1.0.0")]
     WriteZero,
+    /// The underlying storage (typically, a filesystem) is full.
+    ///
+    /// This does not include out of quota errors.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    NoStorageSpace,
+    /// Invalid seek
+    ///
+    /// Seeking was attempted on an open file handle which is not suitable for seeking - for
+    /// example, on Unix, a named pipe opened with `File::new`.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    InvalidSeek,
```

Comment:
This looks like window's [`ERROR_SEEK`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-), [`ERROR_NEGATIVE_SEEK`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-), or [`ERROR_SEEK_ON_DEVICE`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-)

Reply:
`ERROR_SEEK` sounds like a hardware failure.  `ERROR_NEGATIVE_SEEK` would be EINVAL on Unix I think.  `ERROR_SEEK_ON_DEVICE` sounds right.